app/controllers/DraftMetadataChecksResultsController.scala

```diff
@@ -33,16 +34,73 @@ class DraftMetadataChecksResultsController @Inject() (
       for {
         reference <- consignmentService.getConsignmentRef(consignmentId, request.token.bearerAccessToken)
         consignmentStatuses <- consignmentStatusService.getConsignmentStatuses(consignmentId, token)
+        errorType <- if (errorTypeDeterminedFromFile(consignmentStatuses)) draftMetadataService.getErrorType(consignmentId) else Future.successful(FileError.UNKNOWN)
       } yield {
-        Ok(
-          views.html.draftmetadata
-            .draftMetadataChecksResults(consignmentId, reference, getValue(consignmentStatuses, DraftMetadataType), request.token.name)
-        )
-          .uncache()
+        val resultsPage = {
+          // leaving original page for no errors
+          if (getValue(consignmentStatuses, DraftMetadataType).value == "IMPORTED") {
+            views.html.draftmetadata
+              .draftMetadataChecksResults(consignmentId, reference, getValue(consignmentStatuses, DraftMetadataType), request.token.name)
+          } else {
+            if (isErrorReportAvailable(errorType)) {
+              views.html.draftmetadata
+                .draftMetadataChecksWithErrorDownload(
+                  consignmentId,
+                  reference,
+                  getValue(consignmentStatuses, DraftMetadataType),
+                  request.token.name,
+                  actionMessage(errorType),
+                  detailsMessage(errorType)
+                )
+            } else {
+              views.html.draftmetadata
+                .draftMetadataChecksErrorsNoDownload(
+                  consignmentId,
+                  reference,
+                  getValue(consignmentStatuses, DraftMetadataType),
+                  request.token.name,
+                  actionMessage(errorType),
+                  detailsMessage(errorType)
+                )
+            }
+          }
+        }
+        Ok(resultsPage).uncache()
       }
     }
   }
 
+  private def actionMessage(fileError: FileError.FileError)(implicit messages: Messages): String = {
+    val key = s"draftMetadata.validation.action.$fileError"
+    if (Messages.isDefinedAt(key))
+      Messages(key)
+    else
+      s"Require action message for $key"
+  }
+
+  private def detailsMessage(fileError: FileError.FileError)(implicit messages: Messages): String = {
+    val key = s"draftMetadata.validation.details.$fileError"
+    if (Messages.isDefinedAt(key))
+      Messages(key)
+    else
+      s"Require details message for $key"
+  }
+
+  private def isErrorReportAvailable(fileError: FileError.FileError): Boolean = {
+    fileError match {
+      case FileError.SCHEMA_VALIDATION => true
+      case _                           => false
+    }
+  }
+
+  private def errorTypeDeterminedFromFile(statuses: List[GetConsignment.ConsignmentStatuses]): Boolean = {
+    val draftMetadataProgress = getValue(statuses, DraftMetadataType)
```

Comment:
You can remove this method after making the above changes.

Reply:
looks better without the if and above def